src/smolagents/agents.py

```diff
@@ -541,21 +587,23 @@ def _run(self, task: str) -> Generator[str, None, None]:
                 step_log.duration = step_log.end_time - step_start_time
                 self.logs.append(step_log)
                 for callback in self.step_callbacks:
-                    callback(step_log)
+                    callback(step_log, self)
```

Comment:
This is also a breaking change: it will break the code of users using old callbacks that accept only one argument.

I am proposing a fix.